skbio/tree/_tree.py

```diff
@@ -3451,84 +3926,58 @@ def distance(self, other):
 
-    def _set_max_distance(self):
-        """Propagate tip distance information up the tree.
+            .. versionadded:: 0.6.3
 
-        This method was originally implemented by Julia Goodrich with the
-        intent of being able to determine max tip to tip distances between
-        nodes on large trees efficiently. The code has been modified to track
-        the specific tips the distance is between
-        """
-        maxkey = itemgetter(0)
+        Returns
+        -------
+        float or int
+            The distance between the two most distant tips in the tree.
+        tuple of (TreeNode, TreeNode)
+            The two most distant tips in the tree.
 
-        for n in self.postorder():
-            if n.is_tip():
-                n.MaxDistTips = ((0.0, n), (0.0, n))
-            else:
-                if len(n.children) == 1:
-                    raise TreeError("No support for single descedent nodes")
-                else:
-                    tip_info = [(max(c.MaxDistTips, key=maxkey), c) for c in n.children]
+        See Also
+        --------
+        distance
+        tip_tip_distances
+        compare_tip_distances
 
-                    dists = [i[0][0] for i in tip_info]
-                    best_idx = np.argsort(dists)[-2:]
-                    (tip_a_d, tip_a), child_a = tip_info[best_idx[0]]
-                    (tip_b_d, tip_b), child_b = tip_info[best_idx[1]]
-                    tip_a_d += child_a.length or 0.0
-                    tip_b_d += child_b.length or 0.0
-                n.MaxDistTips = ((tip_a_d, tip_a), (tip_b_d, tip_b))
+        Notes
+        -----
+        If a node does not have an associated branch length, 0 will be used.
 
-    def _get_max_distance_singledesc(self):
-        """Return the max distance between any pair of tips.
+        When a tie is observed among more than one pair of tips, only one pair will be
+        returned. The choice is stable. This often happens when `use_length=False`.
 
-        Also returns the tip names  that it is between as a tuple
-        """
-        distmtx = self.tip_tip_distances()
-        idx_max = divmod(distmtx.data.argmax(), distmtx.shape[1])
-        max_pair = (distmtx.ids[idx_max[0]], distmtx.ids[idx_max[1]])
-        return distmtx[idx_max], max_pair
-
-    def get_max_distance(self):
-        """Return the max tip tip distance between any pair of tips.
-
-        Returns
-        -------
-        float
-            The distance between the two most distant tips in the tree
-        tuple of TreeNode
-            The two most distant tips in the tree
-
-        Raises
-        ------
-        NoLengthError
-            A NoLengthError will be thrown if a node without length is
-            encountered
-
-        See Also
-        --------
-        distance
-        tip_tip_distances
-        compare_tip_distances
+        References
+        ----------
+        .. [1] Mai, U., & Mirarab, S. (2018). TreeShrink: fast and accurate detection
```

Comment:
I'm a bit puzzled by this reference -- this method predates that paper, and the authors didn't contribute the code?

Reply:
Or is it using their implementation?

Reply:
The paper explicitly introduced the notion of "diameter". But I think there should be lots of mentionings of this term prior to that paper. Given your concern, let me remove this citation.